esx_service/utils/kvESX.py

```diff
@@ -175,24 +185,26 @@ def load(volpath):
     except IOError:
         return None
 
-    kvDict = json.load(fh)
-
+    kv_str = fh.read()
     fh.close()
-
-    return kvDict
+    if kv_str: 
+       return json.loads(kv_str)
+    else:
+       return None
 
 
 # Save the dictionary to side car.
-def save(volpath, kvDict):
+def save(volpath, kv_dict):
     metaFile = lib.DiskLib_SidecarMakeFileName(volpath, DVOL_KEY)
 
```

Comment:
That's why I'm my original change I was handling the exception here. No, calling code doesn't handle exceptions at all. These modules don't define exceptions as yet for any of these APIs, which may be a todo via a separate issue.